File: examples/R/example_function.R
test <- function(x, y, ...) {
  op <- par(no.readonly=TRUE)
  on.exit(par(op))
  args <- list(...)
  if (any(is.na(x) | is.na(y))) return(NULL)
  x+y
}

test(c(7,3), c(3,7))
test(c(7,3), c(3,NA))

# short circuit evaluation
isTRUE                  # reference to function
isTRUE("hallo, welt")   # call to function
isTRUE(NA)

File: examples/Python/example_function.py
import numpy as np

def test(x, y, *args):
    if np.any(np.isnan(x)) or np.any(np.isnan(y)):
        return None
    return x + y

# Test the function
print(test(np.array([7, 3]), np.array([3, 7])))  # Output: [10 10]
print(test(np.array([7, 3]), np.array([3, np.nan])))  # Output: None

# Short-circuit evaluation
def is_true(x):
    return (isinstance(x, bool) and (x is True))
print(is_true)                  # reference to function     
print(is_true(True))            # call to function
print(is_true("hallo, welt")) 
print(is_true(None))
